pkg/network/http/driver_interface.go

```
@@ -0,0 +1,238 @@
+				log.Infof("Error reading http transaction buffer: %v", err)
+				continue
+			}
+
+			batchSize := bytesRead / transactionSize
+			transactionBatch := make([]driver.HttpTransactionType, batchSize)
+
+			for i := uint32(0); i < batchSize; i++ {
+				transaction := (*driver.HttpTransactionType)(unsafe.Pointer(&buf.Data[i*transactionSize]))
+				deepCopyTransactionData(&transactionBatch[i], transaction)
+			}
+
+			di.dataChannel <- transactionBatch
+
+			err = driver.StartNextRead(di.driverHTTPHandle.Handle, buf)
+			if err != nil && err != windows.ERROR_IO_PENDING {
+				log.Infof("Error starting next http transaction read: %v")
+			}
+		}
+	}()
+}
+
+func iocpIsClosedError(err error) bool {
+	if err == nil {
+		return false
+	}
+	// ERROR_OPERATION_ABORTED or ERROR_ABANDONED_WAIT_0 indicates that the iocp handle was closed
+	// during a call to GetQueuedCompletionStatus.
+	// ERROR_INVALID_HANDLE indicates that the handle was closed prior to the call being made.
+	return err == syscall.Errno(windows.ERROR_OPERATION_ABORTED) ||
+		err == syscall.Errno(windows.ERROR_ABANDONED_WAIT_0) ||
+		err == syscall.Errno(windows.ERROR_INVALID_HANDLE)
+}
+
+func deepCopyTransactionData(dest, src *driver.HttpTransactionType) {
+	dest.Tup.Saddr = src.Tup.Saddr
+	dest.Tup.Daddr = src.Tup.Daddr
+	dest.Tup.Sport = src.Tup.Sport
+	dest.Tup.Dport = src.Tup.Dport
+	dest.Tup.Protocol = src.Tup.Protocol
+	dest.Tup.Family = src.Tup.Family
+	dest.Tup.Pid = src.Tup.Pid
+
+	dest.RequestStarted = src.RequestStarted
+	dest.ResponseLastSeen = src.ResponseLastSeen
+	dest.RequestMethod = src.RequestMethod
+	dest.ResponseStatusCode = src.ResponseStatusCode
+	dest.RequestFragment = src.RequestFragment
+}
+
+func (di *httpDriverInterface) flushPendingTransactions() {
+	err := windows.DeviceIoControl(di.driverHTTPHandle.Handle,
+		driver.FlushPendingHttpTxnsIOCTL,
+		nil, uint32(0), nil, uint32(0), nil, nil)
+	if err != nil {
+		log.Warnf("Failed to flush pending http transactions: %v", err)
```

I'd return the error here and log where it is called.